web/components/roomConfig/RoomConfig.tsx

```diff
@@ -0,0 +1,124 @@
+import { h, FunctionComponent, Fragment } from "preact";
+import { useState } from "preact/hooks"
+import { ListItem } from "../ListItem";
+import BridgeAPI from "../../BridgeAPI";
+import ErrorPane from "../ErrorPane";
+import style from "./RoomConfig.module.scss";
+import { GetConnectionsResponseItem } from "../../../src/provisioning/api";
+
+
+export interface ConnectionConfigurationProps<SConfig, ConnectionType extends GetConnectionsResponseItem, ConnectionState> {
+    serviceConfig: SConfig;
+    onSave: (newConfig: ConnectionState) => void,
+    existingConnection?: ConnectionType;
+    onRemove?: () => void,
+    api: BridgeAPI;
+}
+
+interface IRoomConfigProps<SConfig, ConnectionType extends GetConnectionsResponseItem, ConnectionState> {
+    api: BridgeAPI;
+    roomId: string;
+    type: string;
+    headerImg: string;
+    text: {
+        header: string;
+        createNew: string;
+        listCanEdit: string;
+        listCantEdit: string;
+    };
+    connectionEventType: string;
+    listItemName: (c: ConnectionType) => string,
+    connetionConfigComponent: FunctionComponent<ConnectionConfigurationProps<SConfig, ConnectionType, ConnectionState>>;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const RoomConfig: FunctionComponent<IRoomConfigProps<any, any, any>> = function<SConfig, ConnectionType extends GetConnectionsResponseItem, ConnectionState>(props: IRoomConfigProps<SConfig, ConnectionType, ConnectionState>) {
+    const { api, roomId, type, headerImg, text, listItemName, connectionEventType } = props;
+    const ConnectionConfigComponent = props.connetionConfigComponent;
+    const [ error, setError ] = useState<null|string>(null);
+    const [ connections, setConnections ] = useState<ConnectionType[]|null>(null);
+    const [ serviceConfig, setServiceConfig ] = useState<SConfig|null>(null);
+    const [ canEditRoom, setCanEditRoom ] = useState<boolean>(false);
+    if (connections === null) {
+        api.getConnectionsForService<ConnectionType>(roomId, type)
+        .then(res => {
+            setCanEditRoom(res.canEdit);
+            setConnections(res.connections);
+        })
+        .catch(ex => {
+            console.warn("Failed to fetch existing connections", ex);
+            setError("Failed to fetch existing connections");
+        });
+    }
+
+    if (serviceConfig === null) {
+        api.getServiceConfig<SConfig>(type)
+        .then((res) => setServiceConfig(res))
+        .catch(ex => {
+            console.warn("Failed to fetch service config", ex);
+            setError("Failed to fetch service config");
+        });
+
+    }
```

Comment:
:warning: This needs fixing.
Creates an endless loop of network calls on failure.

1. Request faills and `setError` is called.
2. Component re-renders with `connections` and `serviceConfig` being `null`.
3. Request faills and `setError` is called.
4. …

The proper way to make network calls on component creation is:
```
useEffect(() => {
    // make your network calls here.
}, []); // The empty array means that this will only be executed once.
```

Reply:
When changing this you'll also need to change `setConnection(null)` to reload the connections.